internal/graph/check.go

```diff
@@ -263,16 +275,26 @@ func intersection(ctx context.Context, concurrencyLimit uint32, handlers ...Chec
 	}
 
 	if err != nil {
-		return &openfgapb.CheckResponse{Allowed: false}, err
+		return &ResolveCheckResponse{
+			Allowed: false,
+			ResolutionMetadata: &ResolutionMetadata{
```

Comment:
We're dropping the resolution depth here. If we're going to to respond with ResolutionMetadata we should carry/propagate that info along.

Reply:
What value should it have?

Reply:
That's a good question. We haven't discussed what the expectation should be.. What do you and @adriantam and @rhamzeh think?

My thoughts - If we're counting database queries, we should be counting how many queries it took to find an outcome (regardless of if it was a permit or deny decision). It's also important for us to know how many database queries are involved in searching for a path and never finding one, because these kinds of queries are often more intensive on the app as well (it's more costly to search the whole graph space than to short-circuit and return early).

Reply:
Ok so, for the case of `allowed = true` where one path was found, `Depth` (height in the tree) should be the height of that solution in particular.

For the case of `allowed = false`, we probably examined a lot of paths each with their own depth. I guess it should be either the maximum of all those paths, or zero 🤷 

But in any case, let's do this in another PR.